File: Challenge3/JavaScript/Challenge3.js
function PrimeFactors() {
    var totalNumber = 600851475143;

    for (var i = 2; i < totalNumber; i++) {
        if (totalNumber % i == 0) {
            totalNumber = (totalNumber / i);
        }
    }
    return totalNumber;
}

console.log(PrimeFactors());

File: Challenge3/Ruby/Challenge3.rb
def modulus_checker
totalNumber = 600851475143

		(2..totalNumber).map do |x| 

				if totalNumber % x == 0 && x != totalNumber
						totalNumber = totalNumber / x	
				end
				
				if totalNumber == x
						return totalNumber
				end	
		end
end

puts modulus_checker
